### PR TITLE
fix(components): add missing target prop to anchor components

### LINKED ISSUE #16848: [4.0.0-rc.0] ion-button ignores target attribute
# Bug Report

**Ionic version:**
[x] **4.x**

**Current behavior:**
When using `ion-button` with `href` and `target` attributes, `<a>` tag is created but is missing `target` attribute.

**Expected behavior:**
`target`attribute should be passed to the `<a>` element inside `ion-button`

**Steps to reproduce:**
```
<ion-button href="/assets/files/FILE.pdf" target="_blank">
    <ion-icon slot="start" name="cloud-download"></ion-icon>
    OPEN IN NEW TAB
</ion-button>
```

**Ionic info:** 
```
Ionic:

   ionic (Ionic CLI)             : 4.6.0 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.0.0-rc.0
   @angular-devkit/build-angular : 0.11.4
   @angular-devkit/schematics    : 7.1.4
   @angular/cli                  : 7.1.4
   @ionic/angular-toolkit        : 1.2.1

System:

   NodeJS : v11.5.0 (/usr/local/Cellar/node/11.5.0/bin/node)
   npm    : 6.5.0
   OS     : macOS Mojave
```

### LINKED ISSUE #16889: Ionic 4 RC-0 - ion-anchor is showing error
# Bug Report

**Ionic version:**

[x] **4.0.0.rc.0**

**Current behavior:**

ion-anchor tag is showing parsing error in ionic v4 release candidate. But, When I import CUSTOM_ELEMENTS_SCHEMA in my signin module, it is working fine.

**Expected behavior:**

ion-anchor tag is work without adding any CUSTOM_ELEMENTS_SCHEMA in my signin module

**Steps to reproduce:**

Implemented ion-anchor tag in my signin page html. Below i have added my code.

**Related code:**

```
<ion-anchor color="primary" routerDirection="forward" href="/forget">Forget your password?</ion-anchor>
```
**Error**
image

**Other information:**

**Ionic info:** 

image

### PR DESCRIPTION
**Pull request by brandyscarney**
This is an extension of the PR here: https://github.com/ionic-team/ionic/pull/17844

###  Adds the following interfaces:

```
export interface AnchorInterface {
  href: string | undefined;
  target: string | undefined;
  rel: string | undefined;
}

export interface ButtonInterface {
  type: 'submit' | 'reset' | 'button';
  disabled: boolean;
}
```

###  Components that implement the interfaces

**Implements the `AnchorInterface` in the following**:
- `anchor`
- `button`
- `card`
- `fab-button`
- `item-option`
- `item`
- `tab-button`

**Implements the `ButtonInterface` in the following**:
- `back-button`
- `button`
- `card`
- `fab-button`
- `item-option`
- `item`
- `menu-button`
- `segment-button`

###  Component properties added

Adds the following properties to the components listed under them:

**`rel`, `download`, `target`**
- anchor
- button
- card
- fab-button
- item-option
- item
- tab-button

**`disabled`**
- back-button
- menu-button

**`type`**
- back-button
- item-option
- menu-button
- segment-button

fixes #16848 closes #16889

### CODE REVIEW

**Comment by brandyscarney** [comment]
@liamdebeasi could you review this? also do you think it should add the `rel` property? The documentation for target includes a blockquote mentioning usage with it: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a

**Comment by brandyscarney** [comment]
So I had the idea to instead just pass down certain attributes since things like `rel` and `download` are also missing. After talking with @jthoms1 on this more, we think it would be better than setting props for everything. I'm not sure if this will cause issues though. Here's a quick example, of course this would be cleaned up more:

```
render() {
    const attributes = ['download', 'href', 'hreflang', 'referrerpolicy', 'rel', 'target', 'type'];
    let attrs = {};

    if (this.el.hasAttributes()) {
      for (let i = 0; i < this.el.attributes.length; i++) {
        const attr = this.el.attributes[i];
        if ((attributes.indexOf(attr.name) > -1)) {
          console.log('pass down', attr.name, '->', attr.value);
          attrs = {
            ...attrs,
            [`${attr.name}`]: attr.value
          };
        }
      }
    }

    return (
      <a
        {...attrs}
        href={this.href}
      >
        <slot></slot>
      </a>
    );
  }
```

We think this may be good for Stencil to handle.

**Comment by bitflower** [comment]
I like generics in general. 
How would you deal with documentation with this though? It wouldn't auto-generated ?
It also hide the props behind a function. Hm, really a tough one!

**Comment by brandyscarney** [comment]
I'm revisiting this @bitflower. After some further discussion it's out of scope to add the automated properties, so instead I am implementing an interface in the components to make sure they're all getting the missing properties. Let me know if you see any properties I've missed. Also, I'll add you as a co-author on this since I am just merging in the fab-button changes. 🙂

**Comment by brandyscarney** [comment]
@liamdebeasi @manucorporat Can you review this? Besides the `ion-anchor` properties, does everything else look ok?

File: core/src/components/menu-button/menu-button.tsx [hunk 1]
```diff
@@ -27,6 +28,11 @@ export class MenuButton implements ComponentInterface {
    */
   @Prop() mode!: Mode;
 
+  /**
+   * If `true`, the user cannot interact with the card.
```
Comment (liamdebeasi) [inline core/src/components/menu-button/menu-button.tsx]:
```suggestion
   * If `true`, the user cannot interact with the menu button.
```

**Comment by bitflower** [comment]
The interface approach is awesome!

**Comment by barocsi** [comment]
for me the target property is still ignored on ionic 5

**Pull request merged**